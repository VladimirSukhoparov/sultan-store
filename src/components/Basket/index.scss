$black:#111111;
$gray:#3F4E65;
$orange:#FFC85E;
$white:#ffffff;

.basket {
    display: flex;
    flex-direction: column;
    width: 100%;
    justify-content: flex-start;
    padding-top: 50px;
    padding-bottom: 100px;

    &__title {
        font-weight: 500;
        font-size: 40px;
        line-height: 120%;
        text-transform: uppercase;
        color: $black;
        margin: 0;

    }

    &__title2 {
        font-weight: 400;
        font-size: 30px;
        line-height: 150%;
        font-style: normal;
        text-align: center;
        margin: 0;
    }

    &__line {
        display: block;
        opacity: 0.3;
        color: $gray;
        width: 100%;
        margin: 50px 0;
    }

    &__container {
        width: 100%;
        display: flex;
        flex-direction: column;
    }

    &__buy {
        display: flex;
        flex-direction: row;
        justify-content: center;
        align-items: center;
        padding: 10px 19px;
        background: $orange;
        border-radius: 79px;
        width: 192px;
        height: 59px;
        font-weight: 700;
        font-size: 14px;
        line-height: 17px;
        color: $white;
        cursor: pointer;
        border: none;

    }

    &__purchase {
        display: flex;
        align-items: center;
        justify-content: space-between;
    }

    &__fullPrice {
        font-weight: 700;
        font-size: 30px;
        line-height: 130%;
        color: $black;

    }

    #thanks {
        display: none;
    }

    #basketNull {
        display: block;
    }


}