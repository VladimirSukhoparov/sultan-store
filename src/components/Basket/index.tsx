import React, { useRef, useState } from 'react'
import { useLocalStorage } from '../../hooks/useLocalStorage';
import { BasketItem } from '../BasketItem';

import './index.scss'

export const Basket = ({ basket, setBasket }) => {
    const { removeAll } = useLocalStorage()
    const thanks = useRef(null);
    const basketNull = useRef(null)

    let set = new Set()

    let res = basket.reduce((a, c) => {
        let str = JSON.stringify(c)
        if (!set.has(str)) {
            set.add(str)
            return a.concat(c);
        }
        return a
    }, [])

    function deleteAll() {
        removeAll('basket');
        setBasket(JSON.parse(localStorage.getItem('basket')) || []);
        setTimeout(() => { basketNull.current.style.display = 'none' },)
        setTimeout(() => {
            thanks.current.style.display = 'block'
            setTimeout(() => {
                thanks.current.style.display = 'none'
                setTimeout(() => { basketNull.current.style.display = 'block' }, )
            }, 2000);
        });

    }


    return (
        <div className='basket' id='basket'>
            <h1 className='basket__title'>–ö–æ—Ä–∑–∏–Ω–∞</h1>
            <hr className='basket__line' />
            <div className="basket__container">
                {basket.length > 0 ? (
                    <>
                        {res.map((item) => (<BasketItem key={item.barcode} item={item} basket={basket} setBasket={setBasket} />))}
                        <div className='basket__purchase'>
                            < a className='basket__buy' id='btnBuy' onClick={() => { deleteAll() }}>–û—Ñ–æ—Ä–º–∏—Ç—å –∑–∞–∫–∞–∑</a>
                            <span className='basket__fullPrice'>{Math.floor(basket.reduce((acc, cur) => acc + parseFloat(cur.price), 0) * 100) / 100} ‚Ç∏</span>
                        </div>
                    </>) : (
                    <h2 className='basket__title2' ref={basketNull} id='basketNull'>–í–∞—à–∞ –∫–æ—Ä–∑–∏–Ω–∞ –ø—É—Å—Ç–∞ üòî</h2>
                )}
                <h2 className='basket__title2' ref={thanks} id='thanks'>–°–ø–∞—Å–∏–±–æ –∑–∞ –∑–∞–∫–∞–∑ üòä</h2>
            </div>
        </div >
    )
}
