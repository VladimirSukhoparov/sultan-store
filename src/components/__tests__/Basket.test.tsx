import { screen, render, cleanup } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";

import { Basket } from "../Basket";

afterEach(() => {
  cleanup();
});

test("should render basket component", () => {
  render(
    <BrowserRouter>
      <Basket basket={[]} setBasket={function (basket: any): void {}} />
    </BrowserRouter>
  );
  const basketElement = screen.getByTestId("basket");
  expect(basketElement).toBeInTheDocument();
  expect(basketElement).toHaveTextContent("Ваша корзина пуста 😔");
});

test("matches snapshot", () => {
  const basket = renderer
    .create(
      <BrowserRouter>
        <Basket basket={[]} setBasket={function (basket: any): void {}} />
      </BrowserRouter>
    )
    .toJSON();
  expect(basket).toMatchSnapshot();
});
