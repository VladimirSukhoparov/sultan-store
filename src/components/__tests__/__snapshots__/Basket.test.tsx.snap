// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="basket"
  data-testid="basket"
  id="basket"
>
  <h1
    className="basket__title"
  >
    Корзина
  </h1>
  <hr
    className="basket__line"
  />
  <div
    className="basket__container"
  >
    <h2
      className="basket__title2"
      id="basketNull"
    >
      Ваша корзина пуста 😔
    </h2>
    <h2
      className="basket__title2"
      id="thanks"
    >
      Спасибо за заказ 😊
    </h2>
  </div>
</div>
`;
